gnc/sub8_perception/src/sub8_pcl/cv_utils.cc

```diff
@@ -1,12 +1,270 @@
 #include <sub8_pcl/cv_tools.hpp>
 
 namespace sub {
+
 cv::Point contour_centroid(Contour& contour) {
   cv::Moments m = cv::moments(contour, true);
   cv::Point center(m.m10 / m.m00, m.m01 / m.m00);
   return center;
 }
 
+
+cv::MatND smooth_histogram(const cv::MatND &histogram, size_t filter_kernel_size, float sigma){
+	cv::MatND hist = histogram.clone();
+	std::vector<float> gauss_kernel = generate_gaussian_kernel_1D(filter_kernel_size, sigma);
+	size_t histSize = hist.total();
+	size_t offset = (filter_kernel_size - 1) / 2;
+	for (size_t i = offset; i < histSize-offset; i++ )   // Convolve histogram values with gaussian kernel
+	{
+		int sum = 0;
+		int kernel_idx = 0;
+		for (int j = i - offset; j <= i + offset; j++){
+			sum += (hist.at<float>(j) * gauss_kernel[kernel_idx++]);
+	    }
+	    hist.at<float>(i) = sum;
+	}
+	for (int i = 0; i < offset; ++i)  // Pad filtered result with zeroes
+	{
+		hist.at<float>(i) = 0;
+		hist.at<float>(histSize - 1 - i) = 0;
+	}
+	return hist;
+}
+
+
+std::vector<float> generate_gaussian_kernel_1D(size_t kernel_size, float sigma){
```

Comment:
Based on [their implementation](http://www.boost.org/doc/libs/master/boost/math/distributions/normal.hpp) yours might be faster because it assumes 0 mean and skips some frivolous checks.